pkgs/by-name/fr/freecad/package.nix

```
@@ -21,7 +21,7 @@
 , ode
 , opencascade-occt_7_6
 , pkg-config
-, python3Packages
+, python311Packages
```

```suggestion
, python311Packages
, python3Packages ? python311Packages
```

Then you can skip the other modifications.

---

Yes, it would not work because `callPackage` provides `python3Packages`. Your current version is fine.